src/java/org/apache/cassandra/index/sai/disk/v2/sortedterms/SortedTermsReader.java

```diff
@@ -0,0 +1,303 @@
+ * and a lookup of the point id by the term is logarithmic.
+ *
+ * <p>
+ * Because the blocks are prefix compressed, random access applies only to the locating the whole block.
+ * In order to jump to a concrete term inside the block, the block terms are iterated from the block beginning.
+ * Expect random access by {@link Cursor#seekToPointId(long)} to be slower
+ * than just moving to the next term with {@link Cursor#advance()}.
+ * <p>
+ * For documentation of the underlying on-disk data structures, see the package documentation.
+ *
+ * @see SortedTermsWriter
+ * @see org.apache.cassandra.index.sai.disk.v2.sortedterms
+ */
+@ThreadSafe
+public class SortedTermsReader
+{
+
+    private final IndexInput termsData;
+    private final SortedTermsMeta meta;
+    private final FileHandle termsTrie;
+    private final LongValues blockOffsets;
+
+    private final long termsDataStartFP; // file pointer to the start of the terms data
+
+    /**
+     * Creates a new reader based on its data components.
+     * <p>
+     * It does not own the components, so you must close them separately after you're done with the reader.
+     * @param termsData a sequence of prefix-compressed blocks each storing a fixed number of terms
+     * @param termsDataBlockOffsets file containing an encoded sequence of the file offsets pointing to the blocks
+     * @param termsTrie handle to the file storing the trie with the term-to-point-id mapping
+     * @param meta metadata object created earlier by the writer
+     */
+    public SortedTermsReader(@Nonnull IndexInput termsData,
+                             @Nonnull IndexInput termsDataBlockOffsets,
+                             @Nonnull FileHandle termsTrie,
+                             @Nonnull SortedTermsMeta meta) throws IOException
+    {
+        this.termsData = termsData;
+        this.termsDataStartFP = termsData.getFilePointer();
+        this.termsTrie = termsTrie;
+        this.meta = meta;
+
+        ByteArrayIndexInput offsestMetaInput = new ByteArrayIndexInput("", meta.offsetMetaBytes);
+
+        DirectMonotonicReader.Meta offsetsMeta =
+                DirectMonotonicReader.loadMeta(offsestMetaInput, meta.offsetBlockCount, DIRECT_MONOTONIC_BLOCK_SHIFT);
+        RandomAccessInput offsetSlice = termsDataBlockOffsets.randomAccessSlice(0, termsDataBlockOffsets.length());
+        this.blockOffsets = DirectMonotonicReader.getInstance(offsetsMeta, offsetSlice);
+    }
+
+    /**
+     * Returns the point id (ordinal) of the target term or the next greater if no exact match found.
+     * If reached the end of the terms file, returns <code>Long.MAX_VALUE</code>.
+     * Complexity of this operation is O(log n).
+     *
+     * @param term target term to lookup
+     * @throws IOException if a read of the internal on-disk trie fails
+     */
+    public long getPointId(@Nonnull ByteComparable term) throws IOException
```

Comment:
AFAICT this doesn't actually throw an `IOException`

Reply:
good catch, thanks, so now that's even simpler